pkg/api/registry/registry.proto

```diff
@@ -34,25 +34,19 @@ message NetworkServiceEndpoint {
     google.protobuf.Timestamp expiration_time = 5;
 }
 
-service NetworkServiceEndpointRegistry {
-    rpc Register (NetworkServiceEndpoint) returns (NetworkServiceEndpoint);
-    rpc Find (NetworkServiceEndpointQuery) returns (stream NetworkServiceEndpoint);
-    rpc Unregister (NetworkServiceEndpoint) returns (google.protobuf.Empty);
-}
-
 service NetworkServiceRegistry {
-    rpc Register (NetworkService) returns (NetworkService);
-    rpc Find (NetworkServiceQuery) returns (stream NetworkService);
-    rpc Unregister (NetworkService) returns (google.protobuf.Empty);
+    rpc Register (NetworkServiceEntry) returns (NetworkServiceEntry);
+    rpc Find (NetworkServiceQuery) returns (stream NetworkServiceEntry);
+    rpc Unregister (NetworkServiceEntry) returns (google.protobuf.Empty);
 }
 
 message NetworkServiceQuery {
-    NetworkService network_service = 1;
-    bool watch = 2;
+    NetworkServiceEndpoint network_service_endpoint = 1;
```

Comment:
So what would the usage pattern for this look like?  Do we ever have a case where we'd send both in the same query?

Reply:
> Do we ever have a case where we'd send both in the same query?

I think yes, let's imagine that we need to monitor for networkservices and for endpoints(for example for vl3 goals). With the current master version, we need to call `Find` for ns_regitry and for nse_registry to achieve it. With this change, we'll able to do this query in one call.

Reply:
Yeah... but is the logic for handling changes to those the same?  I suspect not.  I suspect that we will have a set of logic that handles NSE changes, and a set of logic that handles NS changes.  Combining the calls makes us combine the logic...